soc/riscv/openisa_rv32m1/soc_context.h

```diff
@@ -24,12 +24,12 @@
 
 /* Initial saved state. */
 #define SOC_ESF_INIT						\
-	0xdeadbaad,						\
-	0xdeadbaad,						\
-	0xdeadbaad,						\
-	0xdeadbaad,						\
-	0xdeadbaad,						\
-	0xdeadbaad
+	0,						\
```

Comment:
If they are all zero, do you even need this #define?

Reply:
Not really, in the Kconfig it is documented as mandatory though
```
config RISCV_SOC_CONTEXT_SAVE
...
	  Enabling this option requires that the SoC provide a
	  soc_context.h header which defines the following macros:
...
	  - SOC_ESF_INIT: structure contents initializer for struct soc_esf
	    state. The last initialized member should not end in a comma.
```
Right now RV32M1_RI5CY is the only soc actually using this feature, so SOC_ESF_INIT could just be removed and it could be hardcoded to zero initialize. If any other core were to use the RISCV_SOC_CONTEXT_SAVE feature a static constant initializer might not fit all use cases anyways, as the init values on the stack could be runtime dependent. Should I just go ahead and replace it with a memset to 0?